src/Tizen.Wearable.CircularUI.Forms.Renderer/CircleWidgetRendererExtension.cs

```diff
@@ -14,38 +14,53 @@
  * limitations under the License.
  */
 
-using System;
+using ElmSharp.Wearable;
 using Xamarin.Forms;
 using Xamarin.Forms.Platform.Tizen;
+using XForms = Xamarin.Forms.Forms;
 
 namespace Tizen.Wearable.CircularUI.Forms.Renderer
 {
     static class CircleWidgetRendererExtension
     {
-        internal static ElmSharp.Wearable.CircleSurface GetSurface(this IVisualElementRenderer renderer)
+        internal static CircleSurface GetSurface(this IVisualElementRenderer renderer)
         {
-            if (null != renderer.Element)
+            Element element = renderer.Element;
+            CircleSurface circleSurface = null;
+            if (element is ICircleSurfaceObject circleElement)
             {
-                Element element = renderer.Element;
-                while (element != null)
+                var provider = circleElement.CircleSurfaceProvider;
+                if (provider != null)
                 {
-                    if (element is CirclePage)
-                    {
-                        var circlePageRenderer = Platform.GetRenderer(element) as CirclePageRenderer;
-                        return circlePageRenderer?.CircleSurface;
-                    }
-                    foreach (var effect in element.Effects)
+                    circleSurface = (CircleSurface)provider.CircleSurface;
+                }
+                else
+                {
+                    circleSurface = GetSurfaceRecursively(element);
+                }
+            }
```

Comment:
```suggestion
            }
            else 
            {
                  circleSurface = GetSurfaceRecursively(element);
            }
```
If an element is not ICircleSurfaceObject, it does not search a CirclePage from parent

Reply:
Yes, I think we don't need to search a circle surface in that case.